mem_concat.go

```diff
@@ -25,7 +26,7 @@ func buildInMemoryConcat(ctx context.Context, client *s3.Client, objectList []*S
 	}
 
 	if estimatedSize < fileSizeMin {
-		data, err := tarGroup(ctx, client, objectList)
+		data, err := tarGroup(ctx, client, objectList, opts.PreservePOSIXMetadata)
```

Comment:
I think it might be beneficial to pass the whole opts *S3TarS3Options object to tarGroup in case we need to pass more options in the future, that way the function type doesn't keep changing/growing.

Reply:
As you wish.